File: memento/src/main/java/com/alexstyl/specialdates/events/peopleevents/PeopleEventsProvider.java
package com.alexstyl.specialdates.events.peopleevents;

import com.alexstyl.specialdates.contact.Contact;
import com.alexstyl.specialdates.date.ContactEvent;
import com.alexstyl.specialdates.date.Date;
import com.alexstyl.specialdates.date.TimePeriod;

import java.util.List;

interface PeopleEventsProvider {
    ContactEventsOnADate fetchEventsOn(Date date);

    List<ContactEvent> fetchEventsBetween(TimePeriod timeDuration);

    List<ContactEvent> fetchEventsFor(Contact contact);

    Date findClosestStaticEventDateFrom(Date date) throws NoEventsFoundException;
}

File: memento/src/main/java/com/alexstyl/specialdates/events/peopleevents/PeopleEventsProvider.kt
package com.alexstyl.specialdates.events.peopleevents

import com.alexstyl.specialdates.contact.Contact
import com.alexstyl.specialdates.date.ContactEvent
import com.alexstyl.specialdates.date.Date
import com.alexstyl.specialdates.date.TimePeriod

interface PeopleEventsProvider {
    fun fetchEventsOn(date: Date): ContactEventsOnADate

    fun fetchEventsBetween(timePeriod: TimePeriod): List<ContactEvent>

    fun fetchAllEventsInAYear(): List<ContactEvent> {
        return fetchEventsBetween(TimePeriod.aYearFromNow())
    }

    fun fetchEventsFor(contact: Contact): List<ContactEvent>

    fun findClosestEventDateOnOrAfter(date: Date): Date?
}
